backend/substrapp/ledger_utils.py

```diff
@@ -7,10 +7,12 @@
 from django.conf import settings
 from rest_framework import status
 from grpc import RpcError
+from substrapp.metrics import get_metrics_client
 
 
 LEDGER = getattr(settings, 'LEDGER', None)
 logger = logging.getLogger(__name__)
+metrics_client = get_metrics_client()
```

Comment:
Why do you use an instance class for a decorator ?
It should be better to have a decorator with class method, no ?

Reply:
@Kelvin-M this is how the statsd lib is exposing the decoration and we inherited from this mechanism